tests/context.go

```diff
@@ -94,6 +94,10 @@ func GetTestContext() *TestContext {
 	return defaultTestContext
 }
 
+func (ctx *TestContext) testContextChanged(t *testing.T) {
+	ctx.req = require.New(t)
+}
+
```

Comment:
I want to noodle on what this section of code addresses. I don't like the idea of even temporarily sharing `t *testing.T` and having a global that is set and shared across tests limits us being able to have to have parallel testing (should our tests start to take too long). 

I had some POC code that I didn't like that allows a `testContext` to generate a child `testContext`. I think the idea has legs, but not that quick and dirty pass.

Leave this for now I suppose and we can work to improve it as time goes on.

Reply:
Ok, makes sense. Looking forward to it :)